src/testing/fuzz.rs

```diff
@@ -0,0 +1,299 @@
+pub use crate::testing::mock::NetworkData;
+use crate::{
+    nodes::NodeIndex,
+    testing::mock::{
+        configure_network, spawn_honest_member, Data, Hasher64, NetworkHook, PartialMultisignature,
+        Signature, Spawner,
+    },
+    Network, NetworkData as ND, SpawnHandle,
+};
+use codec::{Decode, Encode, IoReader};
+use futures::{
+    channel::oneshot::{self, Receiver},
+    StreamExt,
+};
+use futures_timer::Delay;
+use log::{error, info};
+use std::{
+    io::{BufRead, BufReader, Read, Result as IOResult, Write},
+    iter::once_with,
+    time::Duration,
+};
+use tokio::runtime::{Builder, Runtime};
+
+struct ClosureHook<F> {
```

Comment:
This is a repeated layer of abstraction – `NetworkHook` essentially already is this. Since you are only using this for `spy` just implement `spy` as a `NetworkHook` directly.

Reply:
flattened